### PR TITLE
CSHARP-4365: Fix Sandcastle warning regarding Microsoft.Extensions.Logging.Abstraction assembly.

### PR DESCRIPTION
**Pull request by rstam**

### CODE REVIEW

**Review by BorisDog — APPROVED** [review]
LGTM (assuming it just works)

File: build.cake [hunk 1]
```diff
@@ -118,6 +118,7 @@ Task("BuildArtifacts")
                     fileNames.Add("DnsClient.dll");
                     fileNames.Add("MongoDB.Libmongocrypt.dll");
                     fileNames.Add("SharpCompress.dll");
+                    fileNames.Add("Microsoft.Extensions.Logging.Abstractions.dll");
```
Comment (DmitryLukyanov) [inline build.cake:121]: I thought about the same solution, but what I don't understand why it started to happen for microsoft library? Maybe it doesn't matter at this point..

Comment (rstam) [reply build.cake:121]: Isn't this a rather new dependency added for the logging work Boris did?

**Review by DmitryLukyanov — APPROVED** [review]
LGTM. NOTE: I don't understand where we need this change and where not. Previously I thought that we need this change for 3rd party libs that may be not fully correct with some "definitions". But I would expect microsoft library should satisfy any "assumptions" about nugets, so it's a mystery for me

**Comment by rstam** [comment]
> LGTM. NOTE: I don't understand where we need this change and where not. Previously I thought that we need this change for 3rd party libs that may be not fully correct with some "definitions". But I would expect microsoft library should satisfy any "assumptions" about nugets, so it's a mystery for me

It's not part of any target framework. It's a Nuget package.

File: build.cake (hunk 1 above)
Comment (DmitryLukyanov) [reply build.cake:121]: yes, but it's not the only new dependency, we have other ones like `ZstdSharp.Port` or `Microsoft.SourceLink.GitHub`, so it's unclear what is special in nugets from this list

Comment (rstam) [reply build.cake:121]: Maybe it's because nothing from those dependencies appear in any public APIs.

**Comment by DmitryLukyanov** [comment]
> > LGTM. NOTE: I don't understand where we need this change and where not. Previously I thought that we need this change for 3rd party libs that may be not fully correct with some "definitions". But I would expect microsoft library should satisfy any "assumptions" about nugets, so it's a mystery for me
> 
> It's not part of any target framework. It's a Nuget package.

can you clarify? I didn't say about TFs

File: build.cake (hunk 1 above)
Comment (DmitryLukyanov) [reply build.cake:121]: That's good point, maybe you're right!

**Comment by rstam** [comment]
> can you clarify? I didn't say about TFs

I think Sandcastle knows about all the public types in net472. Anything else has to come from a DLL in this list.

File: build.cake (hunk 1 above)
Comment (DmitryLukyanov) [reply build.cake:121]: but same story with `SharpCompress.dll` :)

**Pull request merged**